### PR TITLE
fix: only warn when partition ckpt and DB ckpt mins are out-of-sync

### LINKED ISSUE #2185: DB went down with replay error when hitting a likely datafusion bug
It seems a Datafusion bug while execute a query lead to IOx replay error

__Steps to reproduce:__
Run the following query in DB `844910ece80be8bc_3c0bd4c89186ca89`

```
SELECT Row_number() 
         OVER ( 
           partition BY Date_trunc('minute', time), bucket_id, partition_id 
           ORDER BY time DESC)    AS rn, 
       gauge, 
       Date_trunc('minute', time) AS time_bucket, 
       env 
FROM   storage_usage_bucket_cardinality 
WHERE  Cast(time AS BIGINT) > Cast(Now() AS BIGINT) - 5 * 60 * 1000000000 
       AND time <= Now(); 
```

We will see the following Datafusion internal error:
`Error running remote query: status: Internal, message: "Internal error reading points from database 844910ece80be8bc_3c0bd4c89186ca89: Internal error executing plan: Arrow error: External error: Arrow error: External error: Arrow error: External error: Execution error: Internal error: Unsupported data type in hasher. This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Tue, 03 Aug 2021 18:55:08 GMT"} }`
Data Ticket: https://github.com/apache/arrow-datafusion/issues/821

Then `844910ece80be8bc_3c0bd4c89186ca89` also goes down with following error:

```
./scripts/grpcurl -plaintext localhost:1234 influxdata.iox.management.v1.ManagementService.GetServerStatus
.....
{
        "dbName": "844910ece80be8bc_3c0bd4c89186ca89",
        "error": {
          "message": "cannot load catalog: Cannot build replay plan: Minimum sequence number in partition checkpoint (123964015) is lower than database-wide number (183235313) for partition http_api_request_duration_seconds:2021-08-02 08:00:00 and sequencer 0"
        },
        "state": "DATABASE_STATE_RULES_LOADED"
      },
....
```

Update: the DF fusion bug and the replay error may not be related. After I wiped the catalog and ran the query again, I hit the same DF bug but the DB is still up running.

### PR DESCRIPTION
**Pull request by crepererum**
There are currently a few bugs and semi-understood edge cases that can
lead to this case. So instead of bailing out, just issue a warning.

Closes #2185.

### CODE REVIEW

File: server/src/db/replay.rs [hunk 1]
```diff
@@ -1838,15 +1821,103 @@ mod tests {
...
-                Step::Assert(vec![Check::Partitions(vec![
-                    ("table_1", "tag_partition_by_a"),
-                    ("table_1", "tag_partition_by_b"),
-                ])]),
+                Step::Assert(vec![
+                    Check::Partitions(vec![
+                        ("table_1", "tag_partition_by_a"),
+                        ("table_1", "tag_partition_by_b"),
+                    ]),
+                    Check::Query(
```
Comment (tustvold) [inline server/src/db/replay.rs:1829]: I don't understand what happened to "table_1,tag_partition_by=a bar=10 0" and "table_1,tag_partition_by=b bar=30 0" .

As far as I understand the test they should have been persisted?

**Review by tustvold — APPROVED** [review]
Tentatively approved, but I am very confused by the tests - they look wrong to me?

File: server/src/db/replay.rs (hunk 1 above)
Comment (crepererum) [reply server/src/db/replay.rs:1829]: same timestamp, aka override / dup

Comment (tustvold) [reply server/src/db/replay.rs:1829]: Oh - brain still booting up - yes we discussed this yesterday :facepalm:

Comment (crepererum) [reply server/src/db/replay.rs:1832]: simplified

**Comment by crepererum** [comment]
Test got simplified and should now be easier to understand.

**Review by tustvold — APPROVED** [review]
Much clearer now :smile:

**Pull request merged**